src/scripts/browserDeviceProfile.js

```diff
@@ -1125,6 +1125,24 @@ export default function (options) {
         Conditions: h264CodecProfileConditions
     });
 
+    if (browser.web0s && supportsDolbyVision(options)) {
+        // Disallow direct playing of DOVI media in containers not mp4.
+        // This paired with the "Prefer fMP4-HLS Container" client playback setting enables DOVI playback on webOS.
+        profile.CodecProfiles.push({
+            Type: 'Video',
+            Container: '-mp4',
+            Codec: 'hevc',
+            Conditions: [
+                {
+                    Condition: 'EqualsAny',
+                    Property: 'VideoRangeType',
+                    Value: 'SDR|HDR10|HLG',
```

Comment:
Your regex was fine.
```suggestion
                    Value: hevcVideoRangeTypes.replace(/\|DOVI[^|]*/g, ''),
```

Reply:
I actually like this version more, stemming from your comment _"Only necessary conditions can be left here "_ on the original PR for this block. It makes sense that for a codec profile that talks about what is not allowed, that codec profile will never be the one used for a playback, right?

I assumed this was the flow the server does-
1. see that a codecprofile says (not mp4 + dovi) = no go
2. do the transcode process
3. check codecprofile again to find the base [hevcCodecProfileConditions](https://github.com/viktory36/jellyfin-web/blob/317c8485ceb703e4654b121edc1a92f76b94231d/src/scripts/browserDeviceProfile.js#L1146-L1150) (that we always push) and all conditions satisfied, which is why we didn't require a codecprofile that mentioned `Container: 'mp4'`.

Unless I'm wrong, and the below conditions that are presently a part of the base hevcCodecProfileConditions also need to be a part of the `-mp4` codec profile?
`{
    "Type": "Video",
    "Codec": "hevc",
    "Conditions": [
        {
            "Condition": "NotEquals",
            "Property": "IsAnamorphic",
            "Value": "true",
            "IsRequired": false
        },
        {
            "Condition": "EqualsAny",
            "Property": "VideoProfile",
            "Value": "main|main 10",
            "IsRequired": false
        },
        {
            "Condition": "EqualsAny",
            "Property": "VideoRangeType",
            "Value": "SDR|HDR10|HLG|DOVI",
            "IsRequired": false
        },
        {
            "Condition": "LessThanEqual",
            "Property": "VideoLevel",
            "Value": "183",
            "IsRequired": false
        }
    ]
}`

During my tests on webOS, I found no difference between keeping all the conditions in the `-mp4` profile vs current version - both led to the expected behavior of starting a remux. I also found the `Container: 'mp4'` profile was not necessary.


Are you okay with staying on this version, on the account it's a lot simpler and in my tests gives the same results as before?

Reply:
OK, I got it - it will fail one of the tests. But it looks tricky.

> I assumed this was the flow the server does-

Step 2 goes last (if no direct profile is found):
HEVC in MP4: server tests only big one.
HEVC in MKV/TS: server tests both.
Then "do the transcode process".

Reply:
Ah, I see. 

There is one other issue in this block - for Dovi profile 8 support on webOS, if nothing changes in the server to specify a unique video range type for DOVI+HDR10 (currently these media are reported as just HDR10/HLG), we will need to remove even HDR10/HLG direct play support for non-mp4, on the off-chance that file actually contains dovi data which would need remuxing.
 ```suggestion
                    Value: 'SDR',
```
I'm happy to do that on my setup, but I don't know if we can speak for every webos+DOVI user in this.

Or is there another way I'm not thinking of?

Reply:
I wouldn't fix this in clients - the media should be detected as DoVi8 (aka DOVI+HDR10) on the server side. AFAIK, media info has an extra field for a profile.